File: core/src/ru/myitschool/dcrawler/entities/ai/task/Task.java
package ru.myitschool.dcrawler.entities.ai.task;

import ru.myitschool.dcrawler.entities.Entity;
import ru.myitschool.dcrawler.event.EntityEventAdapter;

/**
 * Created by Voyager on 12.02.2018.
 */
public abstract class Task extends EntityEventAdapter {

    protected Entity entity;
    private boolean started = false;
    private boolean complete = false;

    protected void complete(){
        complete = true;
    }

    public boolean isComplete() {
        return complete;
    }

    public boolean isStarted() {
        return started;
    }

    public void setEntity(Entity entity) {
        this.entity = entity;
    }

    public void start(){
        started = true;
        startTask();
    }

    protected abstract void startTask();
}

File: core/src/com/wolg_vlad/dcrawler/entities/ai/task/Task.kt
package com.wolg_vlad.dcrawler.entities.ai.task

import com.wolg_vlad.dcrawler.entities.Entity
import com.wolg_vlad.dcrawler.event.EntityEventAdapter

/**
 * Created by Voyager on 12.02.2018.
 */
abstract class Task : EntityEventAdapter() {
    var entity: Entity? = null
    var isStarted = false
        private set
    var isComplete = false
        private set

    protected fun complete() {
        isComplete = true
    }

    fun start() {
        isStarted = true
        startTask()
    }

    protected abstract fun startTask()
}